src/raft_server.c

```diff
@@ -825,11 +825,8 @@ int raft_recv_installsnapshot_response(raft_server_t* me_,
     if (0 != e)
         return e;
 
-    if (!r->complete)
-        return 0;
-
     /* The snapshot installation is complete. Update the node state. */
-    if (raft_node_get_match_idx(node) < r->last_idx)
+    if (r->complete && raft_node_get_match_idx(node) < r->last_idx)
```

Comment:
probably have become too rusty on raft protocol, but am wondering if in the r->complete = false case, when the code runs raft_send_appendentries() below, the expectation I guess is that it will always go down the path of invoking raft_send_installsnapshot() - from the raft_send_appendentries() code
    if (next_idx <= log_get_base(me->log))
        return raft_send_installsnapshot(me_, node);

is there any chance it could instead send AE RPC instead of IS RPC?

Reply:
Yes, it can send AE instead of IS, depending on the latest node state, if this IS response is a duplicate that arrives very late, for instance. The aim is to check if something (entries or chunks) can be sent.